src/components/templates/discount-table.tsx

```diff
@@ -0,0 +1,368 @@
+            onClick: (e) => {
+              setDeleteDiscount(discount)
+            },
+            icon: <TrashIcon size={20} />,
+          },
+        ]}
+      >
+        <Table.Cell>
+          <Badge variant="code">
+            <span className="inter-small-regular">{discount.code}</span>
+          </Badge>
+        </Table.Cell>
+        <Table.Cell>{discount.rule.description}</Table.Cell>
+        <Table.Cell>{getDiscountAmount(discount)}</Table.Cell>
+        <Table.Cell>{getDiscountStatus(discount)}</Table.Cell>
+        <Table.Cell>
+          {discount.rule.valid_for.length ? (
+            <span>
+              {discount.rule.valid_for[0].title}
+              <span className="text-grey-40">
+                {discount.rule.valid_for?.length > 1 &&
+                  ` + ${discount.rule.valid_for.length - 1} more`}
+              </span>
+            </span>
+          ) : (
+            <span>All Products</span>
+          )}
+        </Table.Cell>
+        <Table.Cell className="text-right pr-4">
+          {discount.usage_count}
+        </Table.Cell>
+      </Table.Row>
+    )
+  }
+
+  return (
+    <div className="w-full h-full flex flex-col justify-between">
+      <div>
+        <Table
+          filteringOptions={filteringOptions}
+          enableSearch
+          placeholder="Search Discounts"
+          handleSearch={handleDiscountSearch}
+        >
+          <Table.Head>
+            <Table.HeadRow>
+              <Table.HeadCell>Code</Table.HeadCell>
+              <Table.HeadCell>Description</Table.HeadCell>
+              <Table.HeadCell className="w-32">Amount</Table.HeadCell>
+              <Table.HeadCell>Status</Table.HeadCell>
+              <Table.HeadCell>Products</Table.HeadCell>
+              <Table.HeadCell className="text-right pr-4">
+                Redemptions
+              </Table.HeadCell>
+              <Table.HeadCell></Table.HeadCell>
+            </Table.HeadRow>
+          </Table.Head>
+          {!(isLoading || isReloading || !discounts || loading) && (
+            <Table.Body>
+              {discounts.map((d, i) => getTableRow(d, i))}
+            </Table.Body>
+          )}
+        </Table>
+        {(isLoading || isReloading || !discounts || loading) && (
+          <div className="w-full pt-2xlarge flex items-center justify-center">
+            <Spinner size={"large"} variant={"secondary"} />
+          </div>
+        )}
+      </div>
+
+      {deleteDiscount && (
+        <DeletePrompt
+          text={"Are you sure you want to remove this discount?"}
+          heading={"Remove discount"}
+          successText="Discount has been removed"
+          onDelete={() =>
+            Medusa.discounts.delete(deleteDiscount.id).then(() => {
+              searchQuery()
+            })
+          }
+          handleClose={() => setDeleteDiscount(undefined)}
+        />
+      )}
+
+      <div className="flex w-full justify-between inter-small-regular text-grey-50">
```

Comment:
Not quite in alignment with the design. Think it needs `mt-8`:
![image](https://user-images.githubusercontent.com/59018053/150942303-742ad8a8-e2ef-4452-8cbc-d933314ec5c2.png)